File: 676/main.go
package main

import (
	"fmt"
)



/**
 * Your MagicDictionary object will be instantiated and called as such:
 * obj := Constructor();
 * obj.BuildDict(dictionary);
 * param_2 := obj.Search(searchWord);
 */
func main() {
	obj := Constructor()
	obj.BuildDict([]string{"hello","hallo","leetcode"})
	param_2 := obj.Search("hhllo")
	fmt.Println(param_2)	// true
}

type MagicDictionary struct {
	l []string
	// m map[string]bool
}


func Constructor() MagicDictionary {
	return MagicDictionary{}
}


func (this *MagicDictionary) BuildDict(dictionary []string)  {
	this.l = dictionary
}


func (this *MagicDictionary) Search(searchWord string) bool {
	for _, w := range this.l {
		c := 0

		if len(w) != len(searchWord) {
			continue
		}
		for i, ch := range searchWord {

			if byte(ch) != w[i] {
				c++
			}
		}
		if c == 1 {
			return true
		}
	}

	return false
}


/**
 * Your MagicDictionary object will be instantiated and called as such:
 * obj := Constructor();
 * obj.BuildDict(dictionary);
 * param_2 := obj.Search(searchWord);
 */

File: 676/main.rs



struct MagicDictionary {
    l: Vec<String>
}


/**
 * `&self` means the method takes an immutable reference.
 * If you need a mutable reference, change it to `&mut self` instead.
 */
impl MagicDictionary {

    fn new() -> Self {
        return MagicDictionary{l: vec![]}
    }
    
    fn build_dict(&mut self, dictionary: Vec<String>) {
        self.l = dictionary
    }
    
    fn search(&self, search_word: String) -> bool {
        for w in self.l.iter() {
            if w.len() != search_word.len() {
                continue
            }
            let mut cnt = 0;
            for (i, ch) in w.chars().enumerate() {
                let cs = search_word.chars();
                for (j, c) in cs.enumerate() {
                    if i == j && ch != c {
                        cnt += 1;
                    }
                }
            }
            if cnt == 1 {
                return true
            }
        }
        false
    }
}

/**
 * Your MagicDictionary object will be instantiated and called as such:
 * let obj = MagicDictionary::new();
 * obj.build_dict(dictionary);
 * let ret_2: bool = obj.search(searchWord);
 */
fn main() {
    let mut obj = MagicDictionary::new();
    obj.build_dict(vec!["hello".to_string(),"hallo".to_string(),"leetcode".to_string()]);
    let ret_2: bool = obj.search("hhllo".to_string());
    println!("{}", ret_2)
}